<?xml version="1.0"?>
<launch>
    <arg name="use_mockup" default="false"/>
    <group unless="$(arg use_mockup)">
        <node pkg="robothon2023" type="task_board_detector" name="task_board_detector" output="screen" respawn="true">
            <rosparam file="$(find robothon2023)/config/perception_params.yaml" command="load"/>
            <remap from="~input_pointcloud_topic" to="/camera/depth_registered/points" />
            <remap from="~input_image_topic" to="/camera/color/image_raw" />
            <remap from="~camera_info_topic" to="/camera/color/camera_info" />
        </node>
        <node pkg="robothon2023" type="points_of_interest_publisher.py" name="points_of_interest_publisher" output="screen" respawn="true">
            <rosparam file="$(find robothon2023)/config/fixed_transforms.yaml" command="load"/>
            <remap from="~approximate_board_pose" to="/task_board_detector/approximate_board_pose" />
            <remap from="~fixed_board_pose" to="/task_board_detector/fixed_board_pose" />
            <remap from="~board_detector_event_out" to="/task_board_detector/event_out" />
        </node>
    </group>

    <group if="$(arg use_mockup)">
        <node pkg="robothon2023" type="pose_mockup_gui.py"
              name="pose_mockup_gui" output="screen">

        <!-- the cycle rate of the node (in hz) -->
        <param name="loop_rate" type="double" value="10.0"/>

        <!-- Reference frame of the published pose -->
        <param name="reference_frame" type="str" value="base_link"/>

        <remap from="~mock_up_pose" to="/mcr_perception/object_selector/output/object_pose"/>
        </node>
    </group>
</launch>
